usr/src/cmd/utmpd/utmpd.c

```diff
@@ -301,6 +354,8 @@ main(int argc, char *argv[])
 	if ((WTMPXfd = open(WTMPX_FILE, O_RDONLY)) < 0)
 		nonfatal("WARNING: unable to open " WTMPX_FILE " for update.");
 
+	door_server_init();
```

Comment:
Since in this implementation the global zone instance of `utmpd` is responsible for fixing up all the zone, it's the only one that neds to set up the door.

```suggestion
	if (getzoneid() == GLOBAL_ZONEID)
		door_server_init();
```